cytotable/utils.py

```diff
@@ -75,3 +79,38 @@ def _duckdb_with_sqlite() -> duckdb.DuckDBPyConnection:
         LOAD sqlite_scanner;
         """
     )
+
+
+def _cache_cloudpath_to_local(path: Union[str, AnyPath]) -> pathlib.Path:
+    """
+    Takes a cloudpath and uses cache to convert to a local copy
+    for use in scenarios where remote work is not possible (sqlite).
+
+    Args:
+        path: Union[str, AnyPath]
+            A filepath which will be checked and potentially
+            converted to a local filepath.
+
+    Returns:
+        pathlib.Path
+            A local pathlib.Path to cached version of cloudpath file.
+    """
+    if (
+        # check that the filepath includes cloud provider prefix
+        any(
+            str(path).lower().startswith(cloudtype)
+            for cloudtype in ["s3://", "gc://", "az://"]
+        )
+        # check that the path is a file (caching won't work with a dir)
+        and AnyPath(path).is_file()
+        # check that the file is of sqlite type
+        # (other file types will be handled remotely in cloud)
+        and AnyPath(path).suffix.lower() == ".sqlite"
+    ):
+        # incur a read which will trigger caching of the file
+        CloudPath(path).read_bytes()
```

Comment:
Do you have a sense of how long this will take? I presume this will be slow

Reply:
This will vary based on where CytoTable is running (network access, hardware data storage type), the cloud provider (AWS, GCP, etc.), and how the bucket is configured (geolocation, throttling, etc). For AWS S3 locations [`cloudpathlib` leverages `boto3` via the `S3Client`](https://github.com/drivendataorg/cloudpathlib/blob/master/cloudpathlib/s3/s3client.py#L153) to perform downloads.

Wondering aloud: would it be a good idea to provide a caveat about larger cloud-based SQLite data (that performance may vary)?